firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/FirebaseCrashlyticsReportManager.java

```diff
@@ -99,19 +109,31 @@ private void onEvent(
       Throwable event, Thread thread, String eventType, boolean includeAllThreads) {
     final long timestamp = currentTimeProvider.getCurrentTimeMillis() / 1000;
 
-    final CrashlyticsReport.Session.Event capturedEvent =
-        dataCapture.captureEventData(
-            event,
-            thread,
-            eventType,
-            timestamp,
-            EVENT_THREAD_IMPORTANCE,
-            MAX_CHAINED_EXCEPTION_DEPTH,
-            includeAllThreads);
-
     final boolean isHighPriority = eventType.equals(EVENT_TYPE_CRASH);
 
-    reportPersistence.persistEvent(capturedEvent, currentSessionId, isHighPriority);
+    CrashlyticsReport.Session.Event.Builder eventBuilder =
+        dataCapture
+            .captureEventData(
+                event,
+                thread,
+                eventType,
+                timestamp,
+                EVENT_THREAD_IMPORTANCE,
+                MAX_CHAINED_EXCEPTION_DEPTH,
+                includeAllThreads)
+            .toBuilder();
+
+    final String content = logFileManager.getLogString();
```

Comment:
out of curiosity, what's the threading model of the method this code is in? is it always on a single thread?

Reply:
For now, yes. It's all synchronous. My goal is to push threading as far out to the callers as possible.

Reply:
That is, it isn't intended to be thread-safe (yet).